network/src/lib.rs

```diff
@@ -73,6 +73,15 @@ use std::collections::{HashMap, HashSet};
 #[cfg(test)]
 mod tests;
 
+mod cost {
```

Comment:
Shouldn't there also be a positive reputation, when good things happen?

Reply:
Yes. I added this which should be done ASAP: #251